### PR TITLE
add cmake build file

### PR DESCRIPTION
**Pull request by linux-fan-dave**

### CODE REVIEW

File: agent/address.h [hunk 1]
```diff
@@ -60,6 +60,7 @@
 #include <sys/socket.h>
 #include <netinet/in.h>
 #include <arpa/inet.h>
+#include <netdb.h>
```
Comment (pwithnall) [inline agent/address.h]: This change looks unrelated to adding cmake support. Why is it needed? What platform are you building on?

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,125 @@
+project (nice)
```
Comment (pwithnall) [inline CMakeLists.txt]: Why do you want to add cmake support? What platform are you building on? The existing autotools build system should work — if there's a particular platform it doesn't work on, then I think I'd rather fix autotools to work on that platform rather than supporting another build system in parallel.

File: agent/address.h (hunk 1 above)
Comment (linux-fan-dave) [inline agent/address.h]: I compile on ubuntu 16.04 and the addrinfo struct is defined in netdb.h on my platforn. According to the man page this is correct.

File: CMakeLists.txt (hunk 2 above)
Comment (linux-fan-dave) [inline CMakeLists.txt]: Yes, the autotools work (at least on linux). On Windows the supplied sln / proj file do not work quite as well. CMake has the same project file for compilation on windows and linux and obsoletes the sln files on windows.. The project file is less complex than the autoconf / automake project and does more or less the same plattform detection and compiling, and works as well on windows. If this is interesting to be added upstream I could add support for the tests as well.

File: agent/address.h (hunk 1 above)
Comment (pwithnall) [inline agent/address.h]: Right, except `struct addrinfo` is only used in `address.c` (not `address.h`), which already conditionally includes `netdb.h`. Is `HAVE_NETDB_H` not being defined in your `config.h`?

File: CMakeLists.txt (hunk 2 above)
Comment (pwithnall) [inline CMakeLists.txt]: While this might improve the build on Windows, I am worried that it would make the situation worse on other platforms. We would be moving from one bad build system (autotools) to another one which is just as bad (cmake). I would rather leave the build system alone, and just accept patches to improve the `.sln` files. The libnice build changes sufficiently irregularly that it is not much work to maintain the `.sln` files manually.

Could you perhaps provide a patch for the `.sln` files based on the diff between git master and what `cmake` generates? Thanks.

File: agent/address.h (hunk 1 above)
Comment (linux-fan-dave) [inline agent/address.h]: you are right - the problem is **HAVE_NETDB_H** was not included in the config.h for windows which made me not include it to the cmake configure process. I will fix this..

Comment (pwithnall) [inline agent/address.h]: OK, but as I said, switching to cmake is not something we're going to accept as maintainers. It will be less effort, and less churn, to stay with autotools and just ensure the `.sln` files are up to date.

Comment (pwithnall) [inline agent/address.h]: > switching to cmake is not something we're going to accept as maintainers. It will be less effort, and less churn, to stay with autotools and just ensure the `.sln` files are up to date.

That’s still true. I’m not going to merge cmake changes, but will happily merge any updates to the `.sln` files if you want to produce them. :-)

**Comment by pwithnall** [comment]
Closing as we don’t want to switch to cmake. If you could produce updates to the `.sln` files which fix your problems, and submit them in a separate pull request, I will happily merge them. Thanks.
